src/ORM/ResultSetFactory.php

```diff
@@ -32,27 +33,29 @@
 class ResultSetFactory
 {
     /**
-     * Constructor
+     * Create a resultset instance.
      *
-     * @param \Cake\ORM\Query\SelectQuery<T> $query Query from where results came.
      * @param iterable $results Results.
+     * @param \Cake\ORM\Query\SelectQuery<T>|null $query Query from where results came.
      * @return \Cake\ORM\ResultSet<array|\Cake\Datasource\EntityInterface>
      */
-    public function createResultSet(SelectQuery $query, iterable $results): ResultSet
+    public function createResultSet(iterable $results, ?SelectQuery $query = null): ResultSet
```

Comment:
@markstory This is a non BC change since we are changing a public method.

If you want I change preserve BC by accepting iterable and selectquery types for both arguments and swapping them internally if passed in the old order.

Reply:
While it *could* be a problem, I think there are likely very few userland implementations of `ResultSetFactory` and the BC isn't as important here.